org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -113,19 +114,50 @@ public void initializeProjects(final Collection<IPath> rootPaths, IProgressMonit
 	}
 
 	protected void importProjects(Collection<IPath> rootPaths, IProgressMonitor monitor) throws CoreException, OperationCanceledException {
+		resetGradleImportErrorStatus();
 		SubMonitor subMonitor = SubMonitor.convert(monitor, rootPaths.size() * 100);
 		for (IPath rootPath : rootPaths) {
 			File rootFolder = rootPath.toFile();
 			for (IProjectImporter importer : importers()) {
 				importer.initialize(rootFolder);
 				if (importer.applies(subMonitor.split(1))) {
-					importer.importToWorkspace(subMonitor.split(70));
+					// for Gradle importer, we catch and collect CoreExceptions
+					if (importer instanceof GradleProjectImporter) {
+						try {
+							importer.importToWorkspace(subMonitor.split(70));
+						} catch (CoreException e) {
+							this.gradleImportErrorStatus.add(e.getStatus());
+						}
+					// for other importer, we keep the current behavior
+					} else {
+						importer.importToWorkspace(subMonitor.split(70));
+					}
 					if (importer.isResolved(rootFolder)) {
 						break;
 					}
 				}
 			}
 		}
+		for (IProject project : getWorkspaceRoot().getProjects()) {
+			cleanAndSetGradleCompatibilityErrorMarker(project);
+		}
```

Comment:
I suggest to update GradleCompatibilityErrorMarker in GradleProjectImporter directly.

Reply:
https://github.com/eclipse/eclipse.jdt.ls/pull/1965/commits/850e74a9773391c7eba071ceda2d57456acd1924 solves this